Qt: removes html tags from tr calls

marcoagner opened the pull request:
Far from glamorous but somewhat necessary, I think. :)
I'm confident I got every offending tr call but let me know if you see something I've missed.
Also, I preferred to keep important strings without divisions (e.g. `tr("bla") + "<b>" + tr("bla")`) since dividing entire phrases may easily confuse translators and lose context in many languages; so take this into consideration for nits on style.

_Obs.: The comments [starting here](https://github.com/bitcoin/bitcoin/pull/13374#issuecomment-394386399) is what brought me to it and I believe the final objective could be to ban html tags from translations as a whole (from just adding a check on `update-translations.py ` to, who knows, add a test to fail if some `tr()` with html tag(s) is found in case there's a clean way to do it) but that's another thing for now._

MarcoFalke commented on src/qt/bitcoingui.cpp:
```diff
@@ -1147,7 +1147,8 @@ void BitcoinGUI::updateProxyIcon()
             QString ip_port_q = QString::fromStdString(ip_port);
             labelProxyIcon->setPixmap(platformStyle->SingleColorIcon(":/icons/proxy").pixmap(STATUSBAR_ICONSIZE, STATUSBAR_ICONSIZE));
-            labelProxyIcon->setToolTip(tr("Proxy is <b>enabled</b>: %1").arg(ip_port_q));
+            labelProxyIcon->setToolTip(tr("Proxy is %1enabled%2: %3")
```
No need to translate `%3`?

MarcoFalke commented on src/qt/askpassphrasedialog.cpp:
```diff
@@ -43,10 +43,16 @@ AskPassphraseDialog::AskPassphraseDialog(Mode _mode, QWidget *parent) :
-            ui->warningLabel->setText(tr("Enter the new passphrase to the wallet.<br/>Please use a passphrase of <b>ten or more random characters</b>, or <b>eight or more words</b>."));
+            {
+            QString text = tr("Enter the new passphrase to the wallet.");
+            text.append(tr("%1Please, use a passphrase of %2ten or more random characters%3, or %2eight or more words%3.")
```
No need to translate `%1`

practicalswift commented on src/qt/askpassphrasedialog.cpp:
```diff
@@ -43,10 +43,15 @@ AskPassphraseDialog::AskPassphraseDialog(Mode _mode, QWidget *parent) :
     {
         case Encrypt: // Ask passphrase x2
-            ui->warningLabel->setText(tr("Enter the new passphrase to the wallet.<br/>Please use a passphrase of <b>ten or more random characters</b>, or <b>eight or more words</b>."));
+            {
```
nit: Why the added block `{ … }`?

marcoagner replied:
Recommenting so the edits do not clutter it (specially because I switched "declaration" by "initialization", oops). As you can see [here](http://en.cppreference.com/w/cpp/language/switch): `"Because transfer of control is not permitted to enter the scope of a variable, if a declaration statement is encountered inside the statement, it has to be scoped in its own compound statement:"`. So the alternatives I know are this or declaring the variable before the case and I don't have a preference here; so if this is ugly/confusing, it may be better to always have it declared before the case label.

Empact replied:
nit: I prefer this inlined too, ala line 120

marcoagner replied:
I agree but the line was very convoluted like `tr("Enter the new passphrase.") + <br/> + tr("Please, use......").arg("<b>", "</b>");` in case of avoiding leave the unnecessary `<br/>`'s placeholder (%1) for translators. Tried this  as a middle ground.

marcoagner replied:
Meh... thinking it through, there's not enough reason to force this extra block pattern. Much better just breaking the line and using `+` operator, I think. Thx!

Empact commented on src/qt/askpassphrasedialog.cpp:
```diff
@@ -43,10 +43,15 @@ AskPassphraseDialog::AskPassphraseDialog(Mode _mode, QWidget *parent) :
-            ui->warningLabel->setText(tr("Enter the new passphrase to the wallet.<br/>Please use a passphrase of <b>ten or more random characters</b>, or <b>eight or more words</b>."));
+            {
+            QString text = tr("Enter the new passphrase to the wallet.");
+            text.append("<br/>" + tr("Please, use a passphrase of %2ten or more random characters%3, or %2eight or more words%3.")
```
%1, %2 would be more intuitive given there is no %1

Empact commented:
Can you include an explanation as to why the html is being excluded in your commit message, for the historical record?

marcoagner commented:
Thank you, @Empact. Updated.

marcohextor committed:
qt: removes html tags from tr() calls

Removing HTML tags from tr() serves the purpose to facilitate
translators work by making them use placeholders like qt's "%1" instead
of possibly more convoluted HTML tags (e.g. a span tag with styles) and,
more importantly, to avoid translators misbehaviors like using HTML tags
and styles to pass verifications on update-address.py in order to render
Bitcoin addresses and whatnot in the GUI.

jonasschnelli commented:
I tend to NACK here since `%1word` seems less understandable for translators then `<b>word`.

marcoagner commented:
Yes, that makes sense.
I suppose this PR only makes sense if the priority is to ban HTML tags from tr() calls. And, unfortunately, as far as I know Qt has no better-looking built-in way to do this without splitting phrases for every style tag; which would probably make translations even trickier for many languages.

laanwj commented:
NACK. I agree with the premise that using HTML in translation messages should be avoided if possible, However I disagree with this specific solution: %1...%2 is not clearer than using HTML, and more error-prone.

promag commented:
@marcoagner are you aware of any project where your approach is used? or if there is a discussion of the pros and cons elsewhere?

I agree with @jonasschnelli though so NACK..

marcoagner commented:
Given your points, I don't think I have a strong case for now with this specific approach.
Thank you all for the time and thoughtful inputs.

MarcoFalke commented:
@marcoagner You could probably whitelist `<b>` if you wanted to reject html tags?
